tasks/move-granules/index.js

```diff
@@ -49,28 +53,83 @@ function buildGranuleDuplicatesObject(movedGranulesByGranuleId) {
   return duplicatesObject;
 }
 
+/**
+ * Updates granule collection to new collection if necessary
+ * @param {ApiGranuleRecord} granule
+ * @param {CollectionRecord} collection
+ * @returns {}
+ */
+async function updateGranuleCollection(granule, collection) {
+  const collectionId = constructCollectionId(collection.name, collection.version);
+  const updatedGranule = { ...granule };
+  if (granule.collectionId !== collectionId) {
+    updatedGranule.collectionId = collectionId;
+    // TODO api call to update granule in pg here
```

Comment:
Zooming out a bit here. Is an API call what we want here? This task could be run on x number of granules. The API call to update a granule is likely going to be quick but this task could result in e.g. 1k invocations to update all of the granules*.

Why use an API write as opposed to a workflow message write? In other words, is there anything we can do to update the granules in the message payload with the new collection info and then let the standard workflow writes handle updating the datastores? 

That may not be feasible for some reason. It would also still require updating the granule upsert logic. What it would get us (theoretically) is a built-in mechanism for updating several granules at once.

*But of course we can instruct users to use this in a way that makes sense. e.g. small batches of granules.

Reply:
Definitely a good follow-on, @Nnaga1 / you should nail down the reporting approach, possibly have an arch session with the team